### PR TITLE
BSTOR-8 - Change style mechanism when clicking on a brand -- Part 1

### PR DESCRIPTION
**Pull request by Pacane**

### CODE REVIEW

**Comment by jclariviere** [comment]
LGTM

File: client/src/main/java/com/arcbees/beeshop/client/application/ApplicationModule.java [hunk 1]
```diff
@@ -28,6 +28,10 @@ protected void configure() {
         install(new ProductModule());
         install(new WidgetModule());
 
+        bind(BrandChangeHandler.class).asEagerSingleton();
+        bind(CurrentBrand.class).to(CurrentBrandImpl.class);
```
Comment (Chris-V) [inline client/src/main/java/com/arcbees/beeshop/client/application/ApplicationModule.java:32]: Shouldn't it be scoped in singleton?

File: client/src/main/java/com/arcbees/beeshop/client/application/ThemeChangerImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            case GQUERY:
+                return resources.style().gquery();
+            case GSSS:
+                return resources.style().gsss();
+            case GAE_STUDIO:
+                return resources.style().gaestudio();
+            case JUKITO:
+                return resources.style().jukito();
+            default:
+                throw new RuntimeException("Cannot select brand style");
```
Comment (Chris-V) [inline client/src/main/java/com/arcbees/beeshop/client/application/ThemeChangerImpl.java:53]: I'd probably just log an error and select Arcbees by default

File: client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import com.arcbees.beeshop.common.dto.Brand;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HasHandlers;
+
+public class BrandChangedEvent extends GwtEvent<BrandChangedEventHandler> {
+    public static Type<BrandChangedEventHandler> TYPE = new Type<>();
+
+    private final Brand brand;
+
+    public BrandChangedEvent(Brand brand) {
```
Comment (Chris-V) [inline client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java:12]: Reduce the visibility. Package-local should still allow mocking

File: client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.arcbees.beeshop.client.events;
+
+import com.arcbees.beeshop.common.dto.Brand;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HasHandlers;
+
+public class BrandChangedEvent extends GwtEvent<BrandChangedEventHandler> {
+    public static Type<BrandChangedEventHandler> TYPE = new Type<>();
```
Comment (Chris-V) [inline client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java:8]: final

File: client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    public static void fire(Brand brand, HasHandlers source) {
+        source.fireEvent(new BrandChangedEvent(brand));
+    }
+
+    public Brand getBrand() {
+        return brand;
+    }
+
+    public Type<BrandChangedEventHandler> getAssociatedType() {
```
Comment (Chris-V) [inline client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java:24]: Add `@Override`

File: client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        return TYPE;
+    }
+
+    protected void dispatch(BrandChangedEventHandler handler) {
```
Comment (Chris-V) [inline client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java:28]: Add `@Override`

File: client/src/main/java/com/arcbees/beeshop/client/application/ApplicationModule.java [hunk 7]
```diff
@@ -28,6 +28,10 @@ protected void configure() {
...
+        bind(ThemeChanger.class).to(ThemeChangerImpl.class).asEagerSingleton();
```
Comment (Chris-V) [inline client/src/main/java/com/arcbees/beeshop/client/application/ApplicationModule.java:33]: Another way of doing that without an extra eager singleton would be to have a Proxy Event on the root presenter (The ApplicationPresenter here). That would also allow you to split the event listening logic from the UI style assignment between presentation and view.

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java [hunk 8]
```diff
@@ -34,24 +44,131 @@
...
+    @UiField
+    AnchorElement gwtpAnchor;
+    @UiField
+    AnchorElement arcbeesAnchor;
+    @UiField
+    AnchorElement gqueryAnchor;
+    @UiField
+    AnchorElement gsssAnchor;
+    @UiField
+    AnchorElement chosenAnchor;
```
Comment (Chris-V) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java:68]: Do you have plans on abstracting the product registration logic? and also presenters should be created by the presenter and set into a slot.

File: client/src/main/java/com/arcbees/beeshop/client/application/ApplicationModule.java (hunk 1 above)
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/ApplicationModule.java:32]: Yes.

File: client/src/main/java/com/arcbees/beeshop/client/application/ApplicationModule.java (hunk 7 above)
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/ApplicationModule.java:33]: Yes I think I'll do that.

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java (hunk 8 above)
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.java:68]: I have an ongoing refactor for this.

File: client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java (hunk 5 above)
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/events/BrandChangedEvent.java:24]: Damn, that means the template from GWT plugin is missing things.

**Pull request merged**